distributed/protocol/cudf.py

```diff
@@ -1,74 +1,20 @@
+import pickle
 import cudf
+import cudf.groupby.groupby
 from .cuda import cuda_serialize, cuda_deserialize
-from .numba import serialize_numba_ndarray, deserialize_numba_ndarray
 
-
-# TODO:
-# 1. Just use positions
-#    a. Fixes duplicate columns
-#    b. Fixes non-msgpack-serializable names
-# 2. cudf.Series
-# 3. Serialize the index
-
-
-@cuda_serialize.register(cudf.DataFrame)
+# all (de-)serializtion code lives in the cudf codebase
+# here we ammend the returned headers with `is_gpu` for
+# UCX buffer consumption
+@cuda_serialize.register((cudf.DataFrame, cudf.Series, cudf.groupby.groupby._Groupby))
 def serialize_cudf_dataframe(x):
-    sub_headers = []
-    arrays = []
-    null_masks = []
-    null_headers = []
-    null_counts = {}
-
-    for label, col in x.iteritems():
-        header, [frame] = serialize_numba_ndarray(col.data.mem)
-        header["name"] = label
-        sub_headers.append(header)
-        arrays.append(frame)
-        if col.null_count:
-            header, [frame] = serialize_numba_ndarray(col.nullmask.mem)
-            header["name"] = label
-            null_headers.append(header)
-            null_masks.append(frame)
-            null_counts[label] = col.null_count
-
-    arrays.extend(null_masks)
-
-    header = {
-        "is_cuda": len(arrays),
-        "subheaders": sub_headers,
-        # TODO: the header must be msgpack (de)serializable.
-        # See if we can avoid names, and just use integer positions.
-        "columns": x.columns.tolist(),
-        "null_counts": null_counts,
-        "null_subheaders": null_headers,
-    }
+    header, frames = x.serialize()
+    header["is_cuda"] = 1
```

Comment:
Do we need this?  I can see that it is added in all of cudf, cupy, numba, but I'm not sure that it is actually used anywhere.

We an address that in a follow-on PR though.